packages/compiler-core/src/parser.ts

```diff
@@ -865,6 +865,13 @@ function condenseWhitespace(
         node.content = node.content.replace(windowsNewlineRE, '\n')
       }
     }
+    // #9229 innerHTML normalize \r\n or \r into a single \n in the DOM
+    else if (
+      node.type === NodeTypes.INTERPOLATION &&
+      node.content.type === NodeTypes.SIMPLE_EXPRESSION
+    ) {
+      node.content.content = node.content.content.replace(/\\r\\n|\\r/g, '\\n')
+    }
```

Comment:
Do we still need this section of the change? Doesn't the code in `hydration.ts` cover it?

I think this part may cause other problems, e.g.:

```html
<script setup>
import { ref } from 'vue'

const msg = ref('\r')
</script>

<template>
  <h1>{{ msg.startsWith('\r') }}</h1>
</template>
```

This will render `false` because the string in the template has been incorrectly changed.

Reply:
Updated.

Reply:
But why is it needed at all? Isn't it redundant?

Reply:
Yes, I've removed that code.